File: strno.py
import math
from collections import Counter
#function primeFactors() i taken from geeksforgeeks 
#URL: "https://www.geeksforgeeks.org/print-all-prime-factors-of-a-given-number/"
def primeFactors(n):
    a=[]
    while n%2==0:
        a.append(2)
        n=n/2
    for i in range(3,int(math.sqrt(n))+1,2):
        while n%i==0:
            a.append(i)
            n=n/i
    if n>2:
        a.append(n)
    return a

#function isprime() is taken from stackexchange
#URL:"https://stackoverflow.com/questions/1801391/how-to-create-the-most-compact-mapping-n-%E2%86%92-isprimen-up-to-a-limit-n"
def isprime(n):
    """Returns True if n is prime."""
    if n == 2:
        return True
    if n == 3:
        return True
    if n % 2 == 0:
        return False
    if n % 3 == 0:
        return False

    i = 5
    w = 2

    while i * i <= n:
        if n % i == 0:
            return False

        i += w
        w = 6 - w

    return True

t=int(input())
for i in range(0,t):
    x,k=[int(x) for x in input().split()]
    if x<=k:
        print("0")
    elif k==1:
        print("1")
    elif k==2:
        if isprime(x):
            print("0")
        else:
            print("1")

    else:
        c=primeFactors(x)
        #d=dict(Counter(c))
        l=len(c)
        #print(l)
        if k<=l:
            print("1")
        else:
            print("0")

File: strno.cpp
#include<bits/stdc++.h>
using namespace std;
//function prime() is taken from https://www.rookieslab.com/posts/fastest-way-to-check-if-a-number-is-prime-or-not
bool prime(unsigned long int n) {
    
    if (n == 1) {
        return false;
    }

    unsigned long int i = 2;
    // This will loop from 2 to int(sqrt(x))
    while (i*i <= n) {
        // Check if i divides x without leaving a remainder
        if (n % i == 0) {
            // This means that n has a factor in between 2 and sqrt(n)
            // So it is not a prime number
            return false;
        }
        i += 1;
    }
    // If we did not find any factor in the above loop,
    // then n is a prime number
    return true;
}
//primefactors code is taken from https://www.geeksforgeeks.org/print-all-prime-factors-of-a-given-number/
int primefactors(unsigned long  int n)
{   
    std::vector<unsigned long int> my_vec;
    vector<unsigned long int>::iterator ip; 
    int count=0;
    
    while (n % 2 == 0)  
    {  
        my_vec.push_back(2);
        //cout << 2 << " ";  
        n = n/2;  
    }  
  
    // n must be odd at this point. So we can skip  
    // one element (Note i = i +2)  
    for (int i = 3; i <= sqrt(n); i = i + 2)  
    {  
        // While i divides n, print i and divide n  
        while (n % i == 0)  
        {  
            my_vec.push_back(i);  
            n = n/i;  
        }  
    }  
  
    // This condition is to handle the case when n  
    // is a prime number greater than 2  
    if (n > 2)  
        my_vec.push_back(n); 

    // Using std::unique 
    //ip = std::unique(my_vec.begin(), my_vec.begin() + my_vec.size());
    //my_vec.resize(std::distance(my_vec.begin(), ip)); 
    return my_vec.size(); 
}
int main()
{
    unsigned long int i,c,t,n,x,k;
    //int count=1,count1=0;
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cin>>t;
    while(t--)
    {   
        cin>>x>>k;
        if(x<=k)
        {
            cout<<"0\n";
        }
        else if(k==1)
        {
            cout<<"1\n";
        }
        else if(k==2)
        {
            if(prime(x))
            {
                cout<<"0\n";
            }
            else
            {
                cout<<"1\n";
            }
            
        }
        else
        {
            c=primefactors(x);
            if(k<=c)
                cout<<"1\n";
            else
                cout<<"0\n";

        }
        
    }
}